starwars_api/models.py

```diff
@@ -52,20 +72,40 @@ def __repr__(self):
         return 'Film: {0}'.format(self.title)
 
 
-class BaseQuerySet(object):
+class BaseQuerySet(object): 
 
     def __init__(self):
-        pass
+        if self.RESOURCE_NAME == 'people':
+            self.result = api_client.get_people()  
+        elif self.RESOURCE_NAME == 'films':
+            self.result = api_client.get_films()
+        
+        self.objects = self.count()
 
     def __iter__(self):
-        pass
+        return self    
+        
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
+        try:
+            if self.RESOURCE_NAME == 'people':
+                return People(self.result['results'].pop(0))
+            elif self.RESOURCE_NAME == 'films':
+                return Films(self.result['results'].pop(0))
+        except:
+            if not self.result['next']:
+                raise StopIteration()
+            else:
+                self.result = api_client._get_swapi(self.result['next'].replace('http://swapi.co/', ""))
+                if self.RESOURCE_NAME == 'people':
+                    return People(self.result['results'].pop(0))
```

Comment:
I'm in the Monday class, but just wanted to say I really liked your code and in particular your `next` implementation of using pop to chew through the results. I was impressed you were able to do it without needing to have any index variables. Really clean code overall!